### PR TITLE
Feat: add shadcn

### LINKED ISSUE #21: Add ShadCN-UI

### PR DESCRIPTION
**Pull request by AmirHossein-z**

### CODE REVIEW

File: package.json
Comment (AmirHosseinKarimi) [inline package.json:1]: Add every package to the project must be a separate commit. As `build` type.

File: package.json [hunk 1]
```diff
@@ -10,19 +10,24 @@
...
     "@types/react": "^18",
     "@types/react-dom": "^18",
     "autoprefixer": "^10.0.1",
+    "eslint": "^8",
+    "eslint-config-next": "14.0.4",
     "postcss": "^8",
     "tailwindcss": "^3.3.0",
-    "eslint": "^8",
-    "eslint-config-next": "14.0.4"
+    "typescript": "^5"
```
Comment (AmirHosseinKarimi) [inline package.json]: Commit should not contain multiple unrelated changes.
Changes on this file should be several commits.
1. `refactor: sort package.json lines`
2. `build: add class-variance-authority package`
3. `build: add clsx package`
4. `build: add lucide-react package`

Every commit must contains the change of `package.json` and `pnpm-lock.yaml` files, which affected by adding specific package to the project.

File: src/app/globals.css
Comment (AmirHosseinKarimi) [inline src/app/globals.css:1]: It must be a standalone commit.
e.g: as `fix: remove unused default styles from global.css`

**Review by AmirHosseinKarimi — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (AmirHossein-z) [reply package.json]: Ok.I remembered it.but this dependencies will be installed when installing shadcn automatically.

File: src/app/globals.css
Comment (AmirHossein-z) [reply src/app/globals.css:1]: This is also automatically configures with shadcn installation

File: package.json (hunk 1 above)
Comment (AmirHosseinKarimi) [reply package.json]: Oh, I did not know about this. So it's okay to bundle all package installations in one commit.

File: src/app/globals.css
Comment (AmirHosseinKarimi) [reply src/app/globals.css:1]: Okay. But it should be in a separate commit. Package installation and configuration are two different actions.
The whole action is bundled in one branch but every change should be in a separate commit.

**Comment by AmirHosseinKarimi** [comment]
@AmirHossein-z I posted some comments on the conversations and then resolved them to merge the PR.
But you can still check the comments above by expanding resolved conversations.

**Review by AmirHosseinKarimi — APPROVED** [review]
